WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.swift

```diff
@@ -988,11 +1002,11 @@ class AbstractPostListViewController : UIViewController, WPContentSyncHelperDele
         }
     }
 
-    func keyForCurrentListStatusFilter() -> String {
+    class func keyForCurrentListStatusFilter() -> String {
         fatalError("You should implement this method in the subclass")
     }
 
-    func currentFilterIndex() -> Int {
+    class func currentFilterIndex() -> Int {
```

Comment:
> I changed these to class methods since they need to be accessed elsewhere for preloading. I didn't want to make an instance of PostListViewController just to access the current filter settings.

Agreed on the principle.

However, when tracking the calls to this method I noticed the only place making such a call without having an instance of the VC is [this one](https://github.com/wordpress-mobile/WordPress-iOS/pull/5655/files#diff-2c22ee07f415c85bb9dc440e62c7c008R544).  (Correct me if I'm wrong, that was a quick search).

That place is asking for the filter to obtain the list of statuses... while it could instead just get the list of statuses directly (which can be ok as a static method).

> I agree that having a bunch of class methods on a VC is ugly, and definitely agree that the setter should correspond. I'd really like to move them out of the VC entirely, as they should be in business logic somewhere, like a model or service. Thoughts on moving them to PostService instead?

Regarding where to place these methods, I'd suggest to keep things simple for the purpose of your PR, to try and reduce the amount of methods that were changed to `class`, and also make sure that getters and setters are on the same page too.

Having the interaction with `NSUserDefaults` in a different class could be a good idea, but I'd suggest not trying to bring that into this PR, as it would be bringing it out of it's scope a bit too much.